File: 679 Dropping Balls/solution.c
#include <stdio.h>

int main() {
    int cases;
    scanf("%d", &cases);
    while(cases--) {
        int D, I;
        scanf("%d %d", &D, &I);
        int ans = 1;
        while(D-- > 1) {
            if(I & 1) {
                ans <<= 1;
                I = (I + 1) >> 1;
            } else {
                ans = (ans << 1) + 1;
                I >>= 1;
            }
        }
        printf("%d\n", ans);
    }
	return 0;
}

File: 679 Dropping Balls/solution.java
import java.io.*;

public class Main {
	public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StreamTokenizer st = new StreamTokenizer(br);
		st.nextToken();
		int cases = (int)st.nval;
        StringBuilder output = new StringBuilder();
        while(cases-- > 0) {
            st.nextToken();
		    int D = (int)st.nval;
		    st.nextToken();
		    int I = (int)st.nval;
		    int ans = 1;
		    while(D-- > 1) {
                if((I & 1) == 1)
                    ans <<= 1;
                else
                    ans = (ans << 1) + 1;
                I = I + (I & 1) >> 1;
            }
            output.append(ans).append('\n');
        }
        System.out.print(output);
	}
}
